### PR TITLE
feat: use tenant authentication

### PR DESCRIPTION
**Pull request by mrcpvn**

### CODE REVIEW

**Comment by WW-CICD** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: pr-CodeBuild-WWTelematicsTenantService-CodeBuildProject
* Commit ID: 493080515815b185f0ea57db7259e17b4bf6487b
* Result: SUCCEEDED
* [Build Logs](https://tlboygrlvi.execute-api.us-east-1.amazonaws.com/Prod/buildlogs?key=28c94c81-4fc6-4c5d-8a7c-d53d0ab3e523%2Fbuild.log) (available for 10 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/WWTelematics.TenantService/serverless.template [hunk 1]
```diff
@@ -121,32 +120,48 @@ Resources:
             Environment:
               Variables:
                 TelematicsNotification__SnsTopicArn: !Ref TenantSnsTopic
+                TelematicsAuthentication__TenantAuthenticationFunctionName: !Ref TenantAuthenticationFunction
```
Comment (davidw-foxtrax) [inline src/WWTelematics.TenantService/serverless.template:123]: I think it would be okay to put this in Globals

Comment (mrcpvn) [reply src/WWTelematics.TenantService/serverless.template:123]: Putting this in Globals will create a circular dependency and the deployment will fail.

Comment (davidw-foxtrax) [reply src/WWTelematics.TenantService/serverless.template:123]: OK, because the function in question is in the same template.  That makes sense.

Comment (mrcpvn) [reply src/WWTelematics.TenantService/serverless.template:123]: correct, the same issue occurs in the client service.

**Pull request merged**